channels/ON-CHAIN.md

```diff
@@ -292,23 +299,28 @@ Serialization defined [here](../serializations.md#channel-slash-transaction)
 - `nonce`: taken from the `from`'s account
 - `fee`
 
-Proof of inclusion represents the internal channel state. At the
-bare minimum it has to include all accounts and their balances but can also
-include contracts and contract calls.
+Proof of inclusion represents the internal channel's state. It has to include both participant
+accounts and their balances.
+If there are any contracts in the channel and those have balances of their own, they are
+not provided in the proof of inclusion but they are rather to be force-pushed in
+subsequent transactions. It is up to participants to decide if they want to
+post them at all. Thus the accumulative balances of the accounts in
+the slash transaction can be lower than the channel balance persisted
+on-chain.
 
 Payload is a valid transaction that has:
 * `state_hash` equal to the proof of inclusion's root hash
 * `channel_id` being the same as the transaction `channel_id`
-* `round` being greater or equal to the last on-chain `round` for that channel id
+* `round` being greater than the last on-chain `round` for that channel id. If
+  the last on-chain transaction had been produced by forced progress, the
+  `round` can also be equal to it. In this case the co-signed payload overwrites
+  the on-chain produced signle-signed force progress
```

Comment:
single-signed?

Reply:
Forcing progress is indeed a single-signed transaction. That's why if there is an off-chain state with the same `round` number and this off-chain transaction is co-signed - the co-signed transaction wins.

Please check [my comment above](https://github.com/aeternity/protocol/pull/180#discussion_r214031361)

Example of this co-signed vs single signed would be:
We have Alice and Bob. They share a channel and a contract in its state `Contract1`. The channel had reached `round=42`. This means that they've both signed a state with some `state_hash` and a `round=42`. Restrospectivly - Bob is particularly unhappy with it for some reason. Bob tries to cheat Alice by posting a force progress transaction with some old state with a `round=41`. Assuming that malicious Bob is not stupid and Bob produces a valid force progress. At this point there is no way that the blockchain knows he had posted some old state - `round=41` seems reasonable. Thus on-chain force progress passes and a new `round=42` is being produced on-chain. Now we have 2 channel's states both with  `round=42`: one is the co-signed they've both agreed upon and that makes Bob unhappy and there is the on-chain produced by a unilaterally forced progress by Bob. If this happens - Alice only has to provide the co-signed state with `round=42` (or a later one for that matter) and Bob's cheating attempt is invalidated.